src/perform_step/adams_bashforth_moulton_perform_step.jl

```diff
@@ -1326,3 +1326,138 @@ end
     f(k4,u,p,t+dt)
   end
 end
+
+# VCABM
+
+function initialize!(integrator,cache::VCABMConstantCache)
+  integrator.fsalfirst = integrator.f(integrator.uprev, integrator.p, integrator.t) # Pre-start fsal
+  integrator.kshortsize = 2
+  integrator.k = typeof(integrator.k)(integrator.kshortsize)
+
+  # Avoid undefined entries if k is an array of arrays
+  integrator.fsallast = zero(integrator.fsalfirst)
+  integrator.k[1] = integrator.fsalfirst
+  integrator.k[2] = integrator.fsallast
+end
+
+@muladd function perform_step!(integrator,cache::VCABMConstantCache,repeat_step=false)
+  @unpack t,dt,uprev,u,f,p = integrator
+  @unpack dts,g,ϕ_n,ϕ_np1,ϕstar_n,ϕstar_nm1,order,max_order = cache
+  k1 = integrator.fsalfirst
+  step = integrator.iter
+  k = order
+  for i = 13:-1:2
+    dts[i] = dts[i-1]
+  end
+  dts[1] = dt
+  ϕ_and_ϕstar!(cache,k1,k)
+  g_coefs!(cache,k+1)
+  u = uprev
+  for i = 1:(k-1)
+      u += dt * g[i] * ϕstar_n[i]
+  end
+  du_np1 = f(u,p,t+dt)
+  ϕ_np1!(cache, du_np1, k+1)
+  u += dt * g[k] * ϕ_np1[k]
+  if integrator.opts.adaptive
+    utilde = dt * (g[k+1]-g[k]) * ϕ_np1[k+1]
+    atmp = calculate_residuals(utilde, uprev, u, integrator.opts.abstol, integrator.opts.reltol,integrator.opts.internalnorm)
+    integrator.EEst = integrator.opts.internalnorm(atmp)
+    if integrator.EEst >= one(integrator.EEst)
```

Comment:
The check for passing is `<=`, so this should be `>`. Any other method where this is done should be changed as well. @YingboMa informing you too in case you did a `>=` anywhere.